pkg/controller/common/util.go

```diff
@@ -1803,7 +1803,7 @@ func ValidateSnapshotCloneProvisioners(ctx context.Context, c client.Client, sna
 }
 
 // GetSnapshotClassForSmartClone looks up the snapshot class based on the storage class
-func GetSnapshotClassForSmartClone(dvName string, targetPvcStorageClassName *string, log logr.Logger, client client.Client) (string, error) {
+func GetSnapshotClassForSmartClone(dvName string, targetPvcStorageClassName, snapshotClassName *string, log logr.Logger, client client.Client) (string, error) {
```

Comment:
Maybe we should define something more deterministic for the `else {` case?
Right now we're just picking the first match, but maybe we can define something like
- StorageProfile
- [Default](https://kubernetes.io/docs/concepts/storage/volume-snapshot-classes/#the-volumesnapshotclass-resource) cluster snap class
- first match in sorted slice

Reply:
Sure we can add the default cluster snap class (if there is one) instead of the first match.

Reply:
Agree but you can only pick the default SnapClass if the provisioner matches.   So the order is something like:
* StorageProfile
* If there is a single snap class matching the provisioner (easy case): Use single matching snap class
* If there are multiple matches and there is a default snap class with matching provisioner: Default cluster snap class
* If there are multiple matches and no matching default snap class: choose first match in sorted slice

Reply:
@aglitke that's exactly what implemented.